packages/wordcount/src/index.js

```diff
@@ -0,0 +1,193 @@
+ * Word counting utility
+ *
+ * Counts the number of words (or other specified type) in the specified text.
+ *
+ * @summary  Count the number of elements in a text.
+ *
+ * @since    2.6.0
+ * @memberof wp.utils.wordcounter
+ *
+ * @param {String}  text Text to count elements in.
+ * @param {String}  type Optional. Specify type to use.
+ *
+ * @param {Object} settings                                   Optional. Key-value object containing overrides for
+ *                                                            settings.
+ * @param {RegExp} settings.HTMLRegExp                        Optional. Regular expression to find HTML elements.
+ * @param {RegExp} settings.HTMLcommentRegExp                 Optional. Regular expression to find HTML comments.
+ * @param {RegExp} settings.spaceRegExp                       Optional. Regular expression to find irregular space
+ *                                                            characters.
+ * @param {RegExp} settings.HTMLEntityRegExp                  Optional. Regular expression to find HTML entities.
+ * @param {RegExp} settings.connectorRegExp                   Optional. Regular expression to find connectors that
+ *                                                            split words.
+ * @param {RegExp} settings.removeRegExp                      Optional. Regular expression to find remove unwanted
+ *                                                            characters to reduce false-positives.
+ * @param {RegExp} settings.astralRegExp                      Optional. Regular expression to find unwanted
+ *                                                            characters when searching for non-words.
+ * @param {RegExp} settings.wordsRegExp                       Optional. Regular expression to find words by spaces.
+ * @param {RegExp} settings.characters_excluding_spacesRegExp Optional. Regular expression to find characters which
+ *                                                            are non-spaces.
+ * @param {RegExp} settings.characters_including_spacesRegExp Optional. Regular expression to find characters
+ *                                                            including spaces.
+ * @param {RegExp} settings.shortcodesRegExp                  Optional. Regular expression to find shortcodes.
+ * @param {Object} settings.l10n                              Optional. Localization object containing specific
+ *                                                            configuration for the current localization.
+ * @param {String} settings.l10n.type                         Optional. Method of finding words to count.
+ * @param {Array}  settings.l10n.shortcodes                   Optional. Array of shortcodes that should be removed
+ *                                                            from the text.
+ *
+ * @return {Number} The number of items counted.
+ */
+export function count( text, type, settings ) {
+	let count = 0;
+	let shortcodes;
+	const defaultSettings  = {
```

Comment:
If there's nothing about these defaults that depends on the function arguments, can we move it to the top of the file so we're not creating new references / executing RegExp `Array#join` on every function call?